### PR TITLE
EN-50442: Rollup might accept filter

### PR DESCRIPTION
**Pull request by chitang**

### CODE REVIEW

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/SoQLAnalysisDepositioner.scala [hunk 1]
```diff
@@ -61,7 +61,8 @@ object SoQLAnalysisDepositioner {
   private def depositionHint[ColumnId,Type](hint: Hint[ColumnId, Type]) = {
     hint match {
       case x@Materialized(_) => x.copy(NoPosition)
-      case x@UniqueOrder(_) => x.copy(NoPosition)
```
Comment (saraid) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/SoQLAnalysisDepositioner.scala:64]: Why was this removed?

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala [hunk 2]
```diff
@@ -278,43 +279,42 @@ class QueryRewriter(analyzer: SoQLAnalyzer[SoQLType]) {
       // still rewrite a query with a window function in if the other clauses all match, however that requires
       // coordination between expression mapping and mapping other clauses at a higher level that isn't implemented,
       // so for now we just forbid it entirely to avoid incorrect rewrites.
-      case fc: FunctionCall if fc.window.nonEmpty || fc.filter.nonEmpty =>
+      case fc: FunctionCall if fc.window.nonEmpty /* || fc.filter.nonEmpty */ =>
```
Comment (saraid) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala:282]: Why keep this comment?

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/SoQLAnalysisDepositioner.scala (hunk 1 above)
Comment (chitang) [reply query-coordinator/src/main/scala/com/socrata/querycoordinator/caching/SoQLAnalysisDepositioner.scala:64]: I accept Marc's comment and decide not implementing unique order via hint.

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala (hunk 2 above)
Comment (chitang) [reply query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryRewriter.scala:282]: I should remove this commented code

**Review by saraid — APPROVED** [review]

**Comment by znep** [comment]
Does this also work with multiple `FILTER` clauses in one query?  Might be good to have a test that validates that, I didn't see one at a quick glance.
